### DISCUSSION TITLE
Integrazione Broadlink RM Mini 3

### DISCUSSION THREAD

**Discussion by cl-7k-01** [indomus/forum#2519] [HUB: Home Assistant]
Buongiorno a tutti, purtroppo non riesco ad integrare il Broandlink RM Mini 3 ad Home Assistant.
Tramite l'app ufficiale (l'unica rimasta perché la app e-control è ormai abbandonata dal 2018..) riesco a connettere il dispositivo alla mia rete wifi (mesh). In molti sono riusciti ad integrare il RM Mini 3 utilizzando l'ultima app disponibile.
Ho riservato per il Mini 3 un indirizzo IP tra le impostazioni dei router mesh.
Al mini 3 viene correttamente assegnato l'indirizzo IP riservato ad ogni connessione (ho verificato con più avvii e riavvii), e si collega senza problemi alla rete wifi.
Quando provo ad integrarlo in HA invece mi blocco subito al primo passaggio: nel campo HOST inserisco l'IP corretto ma la connessione non avviene.
Ho tentato, come suggerito nelle guide e da altri utenti, anche di togliere il flag all'opzione "dispositivo di blocco", all'interno della app Android broadlink, ma nulla.
Naturalmente ho seguito la guida interrompendo la configurazione subito dopo la connessione del mini 3 con il Wifi. Ho fatto diversi tentativi, anche completando la configurazione da app (per verificare se almeno in quel modo il mini 3 funzionava).
L'RM mini 3 funziona ed è connesso alla rete wifi, ma HA non lo vede.
Sbaglio qualcosa? Cosa mi resta da provare? Il problema è forse la rete mesh?
Home Assistant è connesso via ethernet con indirizzo 192.168.**1**.xxx mentre il Broadlink è connesso alla rete via wifi con indirizzo 192.168.**68**.xxx. Può essere questo il problema?

Ho letto questa discussione: [https://github.com/indomus/forum/discussions/582](url) , può essere necessario qualcosa del genere?

Home Assistant 2023.6.3
Supervisor 2023.06.4
Operating System 10.3
Frontend 20230608.0 - latest

Grazie per l'attenzione
Pier

**Comment by cl-7k-01** [comment]
Come mi ero immaginato:
il problema è nell'indirizzo IP che la mia rete mesh assegna al RM 3 mini.
Ho fatto una prova, disattivando la rete mesh e attivando il WIFI del modem: al Broadlink viene assegnato un indirizzo IP del tipo 192.168.**_1_**.XXX al posto del 192.168.**_68_**.XXX
Così facendo la connessione l'integrazione in Home Assistant avviene senza alcun problema.
Può essere considerato un bug di Home Assistant, o meglio, dell'integrazione Broadlink?
è il caso di segnalare in qualche modo?

**Reply by KC-inDomus** [reply]
non stai facendo meshing in modo corretto, la class edeve rimanere la stessa (192.168.1.0/24)

**Reply by cl-7k-01** [reply]
Ciao, per la rete mesh uso dispositivo Deco di TPLink e non posso fare diversamente. Secondo te sbaglio io o è il mio sistema mesh che non lo permette?

![Immagine WhatsApp 2023-07-02 ore 09 12 08](https://github.com/indomus/forum/assets/4005789/18cc2d33-6437-4b06-83d2-1e73600c37c6)

![Immagine WhatsApp 2023-07-02 ore 09 15 45](https://github.com/indomus/forum/assets/4005789/5f3a2ed4-a70e-47ab-a9b3-f5128dc10381)

**Comment by cl-7k-01** [comment]
Ho fatto un profondo aggiornamento della mia rete cablata in casa. Effettivamente non era impostata nel modo corretto.
Ora la rete è settata "da manuale" (allego schema TP-Link).
Detto ciò, la rete creata da TP-Link (cioè tutta) ha come indirizzo 192.168.68.XXX.
Gli unici dispositivi con un indirizzo diverso sono modem (192.168.1.1) e il router Deco (192.168.68.2).
Ora, tutti i dispositivi cablati o no, Home Assistant compreso, sono sotto rete 192.168.68.XXX.
La rete mesh è corretta.
RM4 Mini continua a non integrarsi. Si collega e assume un indirizzo IP del tipo 192.168.68.XXX (riservato), da app funziona, in Home Assistant no.
L'unico modo per integrarlo è che sia su un IP 192.168.1.XXX.

![Cattura](https://github.com/indomus/forum/assets/4005789/9c4c1e09-97cc-4ee0-b0d6-e7e6b9f6cc26)

**Comment by cl-7k-01** [comment]
Se non ho capito male il traffico generato da e per il broadlink passa per la porta 80.
Non sono un esperto di reti e chiedo il vostro aiuto, se potete e se vi va.
Potrebbe essere che la debba in qualche modo aprire la porta 80? Potrebbe essere che, i Deco abbiano la porta 80 chiusa?
Se così fosse perchè non accade lo stesso con i dispositivi shelly (porta 1883) connessi alla stessa rete mesh? Io non ho aperto la porta 1883 eppure i dispositivi shelly via mqtt non mi danno alcun problema..

**Reply by KC-inDomus** [reply]
è broadlink ad aprire ed esporre la porta 80, tu non devi far nulla - così come DAGLI shelly la porta aperta VERSO la 1883 (aperta ed esposta dal broker) è casuale.

**Reply by cl-7k-01** [reply]
Grazie per la risposta.
Ho spostato la mia attenzione sulla configurazione della rete.
Dopo mille tentativi ho provato così: HA e Deco Principale (che genera la rete Wifi) collegati in ethernet al modem, poi il secondo Deco in backhual wifi con il Deco principale.
I Deco sono ora settati in configurazione Accesso Point e non Router mode (non generano quindi una sottorete).
Così facendo i componenti Shelly funzionano e sono riuscito integrare anche il broadlink. Senza naturalmente agire su porte o quant'altro come infatti dicevi.
Accade però una cosa molto particolare: se accendo lo switch (Tp-link, 16 porte) attualmente collegato via ethernet al Deco principale, cade tutto, nel giro di qualche minuto tutti gli Shelly non sono più disponibili e neppure il Broadlink.
Devo collegare pure lo switch direttamente al modem?
Dello switch ho necessità, ho seminato la casa di prese di rete e mi ritrovo che a funzionare è solo il wifi...
Grazie per la pazienza e l'attenzione.

**Comment by cl-7k-01** [comment]
Ho risolto.
Da oggi la mia configurazione di rete è questa: modem - switch - Deco (2) in modalità AP connessi via cavo Ethernet allo switch.
Ho resettato la rete Wifi Deco e sono ripartito da zero.
Resta il fatto, per me e le mie conoscenze inspiegabile, che, **in configurazione Router Mode (con generazione di sottorete)  e con la "cablatura" impostata come da manuale del produttore, funzioni tutto, tranne l'integrazione del Broadlink, che, per inciso, da app funzionava e si connetteva regolarmente alla wifi.** Non so a chi "dare" la colpa.
Lo sviluppatore punta il dito contro configurazione di rete (firewall del Deco?) e ha assicurato che, anche in sottorete, il Broadlink avrebbe dovuto integrarsi.
Ora, con questa configurazione, che non compare in nessuna pagina di supporta Deco (almeno per il modello da me acquistato), ma che per un ignorante come me sembrava da sempre la più logica, tutto funziona, sia la parte cablata gestita dalla switch sia la parte Wifi gestita dai Deco, tutto sotto una unica rete.
